### ISSUE TITLE
Redo currency system

### ISSUE THREAD

**Issue by UnsignedArduino** [actualdankcoder/erinRewrite#1]
No idea what to do yet, will wait on more input and talk here until we write an "outline" in a PR or something.

Goals so far:

### Obtaining Erin:
- [x] `hourly` command
- [x] `daily` command
- [x] `monthly` command
- [ ] `work` command
- [ ] `pick` command (in chat from `plant`ed)
- [ ] Currency drops from our beloved Erin
### Spending Erin:
- [ ] `shop` command, see [this comment](https://github.com/actualdankcoder/erinRewrite/issues/1#issuecomment-901943731).
- [ ] `send` command
- [ ] `plant` command (in chat for people to get)
- [x] `betflip` command
- [x] `betdice` command
- [x] `betwheel` command
### Other
- [x] `balance` command

**Comment by actualdankcoder** [comment]
I'm in favour of this issue, the currency system is indeed very messy. We need to have:

- A single currency or perhaps an easier approach to it ( there are about 80 currencies rn which are confusing to newer people )
- More Economy Commands
- A way to actually earn currency ( The only way to earn atm is erin drops or hourly gifts )
- Something like stocks, people can invest in it and based on real market data we could evaluate their profits/losses

**Comment by UnsignedArduino** [comment]
What's definitely the most pressing issue: What are we going to call it? Erin coins, with Erin's face on a circular emoji?

**Comment by UnsignedArduino** [comment]
For the way to earn it, I'm thinking of something simple like the `timely` command for Mikasa, I think we should do the same, but we call it `hourly` and also have a `daily` as well which maybe gives about 12 times the amount of hourly.

**Comment by UnsignedArduino** [comment]
Also, will it be pretty common like Dank Memer's coins where you can easily get hundreds of thousands of them bit or "rare" like Mikasa's flowers where having over a million flowers is extremely rare?

**Comment by actualdankcoder** [comment]
> For the way to earn it, I'm thinking of something simple like the `timely` command for Mikasa, I think we should do the same, but we call it `hourly` and also have a `daily` as well which maybe gives about 12 times the amount of hourly.

that is what is currently being used, we need an additional command like "xwork" which would do something the dank memer's work command

**Comment by actualdankcoder** [comment]
and a shop, because simply having currency is boring and there should be something to spend them on

**Comment by UnsignedArduino** [comment]
For the shop, I think it should for now, just be special roles you can buy, and you can add and remove roles to buy from the shop using these commands:
(prefix is assumed to be `x`)
```text
(in config cog)
xshopmanagerole add @Manager  (for example)
xshopmanagerole add @Admin
xshopmanagerole add @Ckyiu
xshopmanagerole remove @Manager (and obv can do without @ so no ping)
xshopmanagerole  (should list all people and roles that can manage the shop)
(in shop cog)
xshop addrole @role 1k  (only @Admin and @Ckyiu for example can run these 3 commands)
xshop changerole @role 2k  (now price of it is 2k)
xshop removerole @role
xshop  (displays all roles you can buy)
xshop buy @role  (you should also be able to do just "role" so no ping)
xshop return @role  (gives back 80% of original cost, so like 1.6k)
```
And then when the user shows the shop we show a paginated thingy which lists all the roles you can buy by price. Not sure if we should do the `return` command though...
But if you also want items to buy like in Dank Memer, then we can do that as well with a big JSON file but we would need to make more emojis for them as well.

**Comment by UnsignedArduino** [comment]
Also why is the gambling commands in a separate cog???

**Comment by UnsignedArduino** [comment]
> that is what is currently being used

Looking through [economy.py](https://github.com/AakashSharma7269/erin/blob/main/src/cogs/economy.py) in the main repo I don't see anything like a `timely` command though

**Comment by actualdankcoder** [comment]
> > that is what is currently being used
> 
> Looking through [economy.py](https://github.com/AakashSharma7269/erin/blob/main/src/cogs/economy.py) in the main repo I don't see anything like a `timely` command though

its called `xclaim` and it works as `xclaim daily` or `xclaim monthly`

**Comment by actualdankcoder** [comment]
> Also why is the gambling commands in a separate cog???

i originally planned to keep it in the same cog but aakash told me to keep it seperate, i think we can keep them in the same cog here

**Comment by actualdankcoder** [comment]
> For the shop, I think it should for now, just be special roles you can buy, and you can add and remove roles to buy from the shop using these commands:
> (prefix is assumed to be `x`)
> 
> ```
> (in config cog)
> xshopmanagerole add @Manager  (for example)
> xshopmanagerole add @Admin
> xshopmanagerole add @Ckyiu
> xshopmanagerole remove @Manager (and obv can do without @ so no ping)
> xshopmanagerole  (should list all people and roles that can manage the shop)
> (in shop cog)
> xshop addrole @role 1k  (only @Admin and @Ckyiu for example can run these 3 commands)
> xshop changerole @role 2k  (now price of it is 2k)
> xshop removerole @role
> xshop  (displays all roles you can buy)
> xshop buy @role  (you should also be able to do just "role" so no ping)
> xshop return @role  (gives back 80% of original cost, so like 1.6k)
> ```
> 
> And then when the user shows the shop we show a paginated thingy which lists all the roles you can buy by price. Not sure if we should do the `return` command though...
> But if you also want items to buy like in Dank Memer, then we can do that as well with a big JSON file but we would need to make more emojis for them as well.

yea that pretty much sums up what i was thinking of

**Comment by UnsignedArduino** [comment]
> > > that is what is currently being used
> > 
> > 
> > Looking through [economy.py](https://github.com/AakashSharma7269/erin/blob/main/src/cogs/economy.py) in the main repo I don't see anything like a `timely` command though
> 
> its called `xclaim` and it works as `xclaim daily` or `xclaim monthly`

Think we should move it to just `daily` and `monthly`? Cause I would have never thought of that lol. Although I prefer more of `hourly` and `daily` cause a month is too long imho...

**Comment by actualdankcoder** [comment]
> > > > that is what is currently being used
> > > 
> > > 
> > > Looking through [economy.py](https://github.com/AakashSharma7269/erin/blob/main/src/cogs/economy.py) in the main repo I don't see anything like a `timely` command though
> > 
> > 
> > its called `xclaim` and it works as `xclaim daily` or `xclaim monthly`
> 
> Think we should move it to just `daily` and `monthly`? Cause I would have never thought of that lol. Although I prefer more of `hourly` and `daily` cause a month is too long imho

yep there's hourly and daily too, the list of codes is in promo.json in the main repo

**Comment by UnsignedArduino** [comment]
I'm going to open a couple of issues to track individual things like "command to print balance" and "daily command," etc.

**Comment by UnsignedArduino** [comment]
Should we deal with separate jobs when we do the work command or just generic jobs like "retype the message" and "click the correct reaction"
